src/fstab-generator/fstab-generator.c

```diff
@@ -936,6 +936,10 @@ static int parse_fstab_one(
                         mount_is_network(fstype, options) ? SPECIAL_REMOTE_FS_TARGET :
                                                             SPECIAL_LOCAL_FS_TARGET;
 
+        /* Setting nofail for /sysroot/usr doesn't make sense so let's just silently ignore it. */
+        if (is_sysroot_usr && FLAGS_SET(flags, MOUNT_NOFAIL))
+                flags &= ~MOUNT_NOFAIL;
+
```

Comment:
this sounds awfully specific to me. i mean, i am not against doing this, but I think if we do this we should do it comprehensively, i.e.:

1. mask both MOUNT_NOFAIL *and* MOUNT_NOAUTO, since neither makes sense
2. do it for both is_sysroot and is_sysroot_usr
3. log about the fact that this was specified like this and that we changed it.

Reply:
4. Drop the masked flags from Options= (I think that leaving it there would be confusing)